src/main/kotlin/org/rust/ide/annotator/RsHighlightingMutableAnnotator.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.psi.PsiElement
+import org.rust.ide.colors.RsColor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsNameIdentifierOwner
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psi.ext.mutability
+import org.rust.lang.core.types.ty.TyReference
+import org.rust.lang.core.types.type
+import org.rust.openapiext.isUnitTestMode
+
+class RsHighlightingMutableAnnotator : Annotator {
+    private val HIGH_PRIORITY_HIGHLIGHTING = HighlightSeverity(
+        "HIGH_PRIORITY_HIGHLIGHTING",
+        HighlightSeverity.INFORMATION.myVal + 3
+    )
+
+    override fun annotate(element: PsiElement, holder: AnnotationHolder) {
+        val ref = when (element) {
+            is RsPath -> element.reference.resolve() ?: return
+            is RsSelfParameter -> element
+            is RsPatBinding -> element
+            else -> return
+        }
+        distinctAnnotation(element, ref, holder)
+    }
+
+    private fun annotationFor(ref: RsElement): Pair<RsNameIdentifierOwner, RsColor>? = when (ref) {
```

Comment:
Looks like that you always return `ref` in result pair. Maybe return only color?